### PR TITLE
Added org.eclipse.kura.log.filesystem.provider bundle

### PR DESCRIPTION
**Pull request by marcellorinaldo**
Brief description of the PR. This PR adds a new bundle that reads device logs from the filesystem.

**Related Issue:** N/A.

**Description of the solution adopted:** The bundle reads system logs from the specified filesystem path and implements the `org.eclipse.kura.log.LogProvider` interface.

**Screenshots:** N/A.

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Comment by MMaiero** [comment]
@marcellorinaldo I have fixed the build that was failing due to a leftover.
But sonar is still failing due to missing tests. Can you please fix that? Thanks

**Comment by MMaiero** [comment]
```
2021-11-13T12:31:03,419 [ConfigurationListener Event Queue] INFO  o.e.k.l.f.p.FilesystemLogProvider - Updated FilesystemLogProvider...
2021-11-13T12:31:03,420 [ConfigurationListener Event Queue] INFO  o.e.k.l.f.p.FilesystemLogProvider - Updated FilesystemLogProvider... Done.
2021-11-13T12:31:03,420 [Thread-8] ERROR o.e.k.l.f.p.FilesystemLogProvider - Unexpected exception in FilesystemLogProvider
java.lang.InterruptedException: sleep interrupted
	at java.lang.Thread.sleep(Native Method)
	at org.eclipse.kura.log.filesystem.provider.FilesystemLogProvider$FileLogReader.run(FilesystemLogProvider.java:106)
2021-11-13T12:31:03,433 [ConfigurationListener Event Queue] INFO  o.e.k.l.f.p.FilesystemLogProvider - Updated FilesystemLogProvider...
2021-11-13T12:31:03,434 [Thread-7] ERROR o.e.k.l.f.p.FilesystemLogProvider - Unexpected exception in FilesystemLogProvider
java.lang.InterruptedException: sleep interrupted
	at java.lang.Thread.sleep(Native Method)
	at org.eclipse.kura.log.filesystem.provider.FilesystemLogProvider$FileLogReader.run(FilesystemLogProvider.java:106)
2021-11-13T12:31:03,437 [ConfigurationListener Event Queue] INFO  o.e.k.l.f.p.FilesystemLogProvider - Updated FilesystemLogProvider... Done.
```

**Comment by marcellorinaldo** [comment]
Do not merge yet, some more fixes needed.

File: kura/test/org.eclipse.kura.log.filesystem.provider.test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    <packaging>eclipse-test-plugin</packaging>
+
+    <properties>
+        <kura.basedir>${project.basedir}/../..</kura.basedir>
+        <sonar.coverage.jacoco.xmlReportPaths>${project.build.directory}/site/jacoco-aggregate/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
+    </properties>
+
+    <build>
+        <plugins>
+			<plugin>
```
Comment (MMaiero) [inline kura/test/org.eclipse.kura.log.filesystem.provider.test/pom.xml]: Formatting seems odd

**Review by MMaiero — APPROVED** [review]

File: kura/test/org.eclipse.kura.log.filesystem.provider.test/pom.xml (hunk 1 above)
Comment (marcellorinaldo) [reply kura/test/org.eclipse.kura.log.filesystem.provider.test/pom.xml]: Corrected.

**Comment by MMaiero** [comment]
We are just a bit over 1000 changed lines. But considering that we are adding also the licensing, we should be good with the Eclipse limits.

**Pull request merged**